### ISSUE TITLE
"Connecting to service" timeout

### ISSUE THREAD

**Issue by chandanvp** [KieronQuinn/DarQ#40]
Hello dev, I have my phone configured to have personal and work profiles by the use of Shelter app. Darq app works good in personal profile. But when installed in work profile, even after granting root permissions, it keeps saying "Connecting to service". After a while, says timeout and to reboot my device. I did that also. No luck here.

I don't know why this issue is happening, but most of the apps I use on work profile never behaved like this.

**Comment by KieronQuinn** [comment]
This is probably related to how Shizuku/Root services work, they'll be cross-profile. If you add an app to DarQ on your personal profile, is it also dark themed on the work profile?

**Comment by chandanvp** [comment]
> This is probably related to how Shizuku/Root services work, they'll be cross-profile. If you add an app to DarQ on your personal profile, is it also dark themed on the work profile? 

Tried this. And yes. If I install an app on both profiles and enable DarQ for the app in personal profile, even in work profile the app would be in dark mode.

But, there are instances where I don't install same apps on both profiles, like, I have isolated Open source apps on personal profile and tracking, proprietary apps on work profile. Most of the Open source apps have dark mode built in, so Darq isn't required.
Some proprietary apps which I use don't have dark theme and I also don't want to install them on personal profile.

Is there a way to get the list of apps on work profile also, when DarQ is installed in personal profile? Because DarQ works fine in personal profile

**Comment by chandanvp** [comment]
Agree. That would be a security concern to show apps from one profile to another. If you are curious to list those, **AppManager** by Muntashirakon(Open source Github Repo) can list all apps. I use it to block trackers on both profiles.

**Comment by Fornball** [comment]
So, is there any progress on this issue? I recently moved all spyware to a work profile and now darq doesnt see them.

**Comment by KieronQuinn** [comment]
libsu recently added multi-user support which should make this possible, but I need to find the time to implement it.

**Comment by heatl0rd** [comment]
I have this issue without using profiles on Android 10. Would there be a reason for that?

**Comment by BLACK4585** [comment]
Yep have the same issue without using profiles. Even the automatic dark mode doesn't work. @KieronQuinn

**Comment by Birraque** [comment]
I also confirm the keeps saying "Connecting to service" when opening. The app still working fine on MIUI 14, Android 13, Kitsune and LSposed.
How can we handle it?
